decorator.rb

```diff
@@ -0,0 +1,26 @@
+require_relative('nameable')
+
+class Decorator < Nameable
+  def initialize(nameable)
+    @nameable = nameable
+    super()
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
```

Comment:
- [ ] Please make sure the `nameable` variable is readable and writable so it's accessible in other classes as well.
  - You can add the nameable variable in `attr_accessor :nameable`.